### DISCUSSION TITLE
Change helm deployed configmap

### DISCUSSION THREAD

**Discussion by davosian** [fluxcd/flux2#1995] [Q&A] [answered]
I am wondering whether the following is possible with flux: I have set up a HelmRelease which in turn deploys a configmap. I now would like to modify the config map with flux and have the daemonset using the configmap to restart. Is this possible and does my approach make sense in the first place?

Note that there is no option provided to override the configmap with values from the helm chart.

**Comment by makkes** [comment, marked as answer]
https://github.com/stakater/reloader might just be what you're looking for. You just need to annotate the DaemonSet and reloader will restart it every time the ConfigMap is modified.

**Reply by davosian** [reply]
@makkes this certainly solves the second part of my question, thanks 🙏 

Can I also tweak a configmap with flux/kustomize while it gets deployed as helmrelease, i.e. I do not want to create the configmap provided by the helm chart but use my own without breaking the helm install process. 

When I manually create the configmap before the helm install, helm will complain that the configmap is not managed by itself and ultimately fails.

**Reply by makkes** [reply]
I would suppose you'd be better off with creating a new ConfigMap and referencing that from the DaemonSet. Does the chart provide that configurability?

**Reply by makkes** [reply]
Modifying resources that are managed by Helm isn't the best idea because the changes might be overwritten at the next upgrade by Flux's helm-controller.

**Reply by davosian** [reply]
The chart maintainer added the option to provide a custom config map. This is probably the cleanest approach. Thanks @makkes for suggesting this.

(yep, modifying helm managed resources is not the best idea as I have experienced myself ;) )
